fsi/fsi.go

```diff
@@ -230,6 +222,36 @@ func (fsi *FSI) Unlink(dirPath, refStr string) error {
 	return fsi.repo.PutRef(ref)
 }
 
+// Remove unlinks the dataset and removes link and low value files associated files
+func (fsi *FSI) Remove(dirPath, refStr string) error {
+	// clean up low value files before running Remove
+	err := removeLowValueFiles(dirPath)
+	if err != nil {
+		log.Errorf("removing low value files: %s", err.Error())
+	}
+	if err := fsi.Unlink(dirPath, refStr); err != nil {
+		return err
+	}
+	// always attempt to remove the directory, ignoring "directory not empty" errors
+	// os.Remove will fail if the directory isn't empty, which is the behaviour
+	// we want
+	if err := os.Remove(dirPath); err != nil && !strings.Contains(err.Error(), "directory not empty") {
+		log.Errorf("removing directory: %s", err.Error())
+	}
+	return nil
+}
+
+// RemoveAll unlinks the dataset and removes all associated files
+func (fsi *FSI) RemoveAll(dirPath, refStr string) error {
+	if err := fsi.Unlink(dirPath, refStr); err != nil {
+		return err
+	}
+	if err := os.RemoveAll(dirPath); err != nil {
```

Comment:
I'm wary of ever calling `RemoveAll`, because it's potentially very destructive, especially when taking unknown / arbitrary input as a parameter. The only places we use it in the codebase right now are for tests that construct temporary directories, and one or two places that are part of rarely used commands, like a user deleting their whole repository. Since we're a data storage system, when it comes to deletes, I'd rather err on the side of being cautious over convenient, and whitelist the things we delete, rather than doing a `RemoveAll`.

In fact, this makes me realize, if a user at some point runs `qri init` in their home directory, and then later run `remove --all --force`, this codepath would try and delete their entire home directory. That alone is reason enough why we shouldn't make `--force` mean "remove even unknown files". The current semantics of `--force` are "delete things even if we get errors, including the working directory not being clean", and I think it should stay that way.

Reply:
Agree on RemoveAll being very agressive and some bad input could wreak havoc.
On the alternative side, doing just RemoveAll and Remove will essentially result in the same behavior, unless Remove no longer removes low value files and it just triggers on `--force` which technically is the purpose of this PR.

Think the above might be a best blend between the two things while remaining cautious.